src/librbd/ImageCtx.cc

```diff
@@ -405,13 +405,15 @@ struct C_InvalidateCache : public Context {
     return flags;
   }
 
-  int ImageCtx::snap_set(string in_snap_name)
+  int ImageCtx::snap_set(cls::rbd::SnapshotNamespace in_snap_namespace,
+			 string in_snap_name)
   {
     assert(snap_lock.is_wlocked());
-    snap_t in_snap_id = get_snap_id(in_snap_name);
+    snap_t in_snap_id = get_snap_id(in_snap_name, in_snap_namespace);
```

Comment:
That's fine -- just want to keep it consistent. The rationale for <namespace, name> is because that is how it's actually stored in ```ImageCtx``` right now (but it could also be switched to <name, namespace>)